protocol/avdl/chat1/remote.avdl

```diff
@@ -4,20 +4,52 @@ protocol remote {
   import idl "github.com/keybase/client/go/protocol/gregor1" as gregor1;
 
   record MessageBoxed {
-    // Only set when returned from the server; on the way up to the
+    // This was not present in most V1 messages.
+    MessageBoxedVersion version;
+
+    // [V1, V2]: Only set when returned from the server; on the way up to the
     // server, they are null.
     union { null, MessageServerHeader } serverHeader;
 
-    // MessageClientHeader is needed by clients to get keys via TLF name.
+    // [V1, V2]: MessageClientHeader is needed by clients to get keys via TLF name.
     // The server needs it as well for sender uid, device id.
     MessageClientHeader clientHeader;
 
+    // V1: Encrypted HeaderPlaintext
+    // V2: Empty
     EncryptedData headerCiphertext;
+
+    // V1: Missing
+    // V2: SignEncrypted HeaderPlaintext
+    SignEncryptedData headerSealed;
+
+    // [V1, V2]: Encrypted BodyPlaintext
     EncryptedData bodyCiphertext;
 
+    // V1: Missing
+    // V2: Public half of the signing key used on headerSealed.
+    //     Used to open and verify headerSealed.
+    //     Must be asserted to belong to the sender when unboxing.
```

Comment:
Whenever I see "don't forget to check X", I always want to get rid of the check, because _someone somewhere_ is going to forget. For example, if instead of providing the key here we just provided a tiny index like "key number 3 for user foo", then the only way to get the key would be to load foo and pull it out of their authentic list of keys. It would be impossible to accidentally verify with a key that wasn't really foo's.

A user's keys are implicitly ordered by when they were delegated in the sigchain, so something like this would be possible today, but out of the scope of this PR for sure.